### PR TITLE
feat: deploy contracts to L1 testnet Sepolia

### PR DESCRIPTION
**Pull request by lesterli**
## Summary

Deploy contracts to L1 Testnet Sepolia

## Test

1. copy `config/deploy_parameters.json.example` to `config/deploy_parameters.json` and change the value of `initialCDKValidiumDeployerOwner` with the deployer address;
2. fill the `environment` in the `docker-compose.yml`;
    
```
cdk-validium-deploy-sepolia:
    - MNEMONIC=
    - INFURA_PROJECT_ID=
 
dac-setup-committee-sepolia:    
    - JSONRPC_HTTP_URL=
    - DEPLOYER_PRIVATE_KEY=
```
3. run `make run-sepolia`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -473,7 +475,8 @@ run: ## Runs a full node
 	$(RUNPOOLDB)
 	$(RUNEVENTDB)
 	$(RUNDACDB)
-	$(RUNL1NETWORK)
+	$(DEPLOYTOSEPOLIANETWORK)
```
Comment (bap2pecs) [inline Makefile:478]: I think we should still keep the default option to use the mock L1 in the quickstart. so `make run` will use the mock L1

we can create a new command `make run-sepolia` that deploy the contract to Sepolia first.

so in the compose file, we should define two containers:

- rename `dac-setup-committee` to `dac-setup-committee-mock-l1`
- create a new one `dac-setup-committee-sepolia`

**Review by bap2pecs — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -10,8 +10,10 @@ DOCKERCOMPOSESTATEDB := cdk-validium-state-db
 DOCKERCOMPOSEPOOLDB := cdk-validium-pool-db
 DOCKERCOMPOSEEVENTDB := cdk-validium-event-db
 DOCKERCOMPOSENETWORK := cdk-validium-mock-l1-network
+DOCKERCOMPOSESEPOLIA := cdk-validium-deploy-sepolia
```
Comment (bap2pecs) [inline Makefile:13]: DOCKERCOMPOSESEPOLIA -> DOCKERCOMPOSEDEPLOYSEPOLIA

File: Makefile [hunk 3]
```diff
@@ -38,9 +40,11 @@ RUNSYNC := $(DOCKERCOMPOSE) up -d $(DOCKERCOMPOSEAPPSYNC)
 RUNETHTXMANAGER := $(DOCKERCOMPOSE) up -d $(DOCKERCOMPOSEAPPETHTXMANAGER)
 RUNGRAFANA := DOCKERGID=`stat -c '%g' /var/run/docker.sock` $(DOCKERCOMPOSE) up -d $(DOCKERCOMPOSEGRAFANA)
 
+DEPLOYTOSEPOLIANETWORK := $(DOCKERCOMPOSE) up -d $(DOCKERCOMPOSESEPOLIA)
```
Comment (bap2pecs) [inline Makefile:43]: RUNDEPLOYSEPOLIA := $(DOCKERCOMPOSE) up -d $(DOCKERCOMPOSESEPOLIA)

File: docker-compose.yml [hunk 4]
```diff
@@ -338,6 +338,21 @@ services:
       - "full"
       - "--rpc.allow-unprotected-txs"
 
+  cdk-validium-deploy-sepolia:
+    container_name: cdk-validium-deploy-sepolia
+    image: snapchain/sepolia-cdk-validium-contracts
```
Comment (bap2pecs) [inline docker-compose.yml:343]: image: snapchain/cdk-validium-contracts

File: docker-compose.yml [hunk 5]
```diff
@@ -361,9 +376,9 @@ services:
       - "-c"
       - "/app/cdk-data-availability run --cfg /app/config.toml"
 
-  dac-setup-committee:
+  dac-setup-committee-mock-l1:
     container_name: dac-setup-committee
-    image: snapchain/dac-setup-committee
+    image: snapchain/sepolia-cdk-validium-contracts
```
Comment (bap2pecs) [inline docker-compose.yml:381]: `image: snapchain/cdk-validium-contracts`

File: docker-compose.yml [hunk 6]
```diff
@@ -375,6 +390,30 @@ services:
...
+        condition: service_completed_successfully
+      cdk-validium-data-availability:
+        condition: service_started
+    volumes:
+      - deploy-output:/shared
+    environment:
+      - JSONRPC_HTTP_URL=https://sepolia.infura.io/v3/<deployer-infura-project-id>
+      - DAC_URL=http://cdk-validium-data-availability:8444
+      - DAC_ADDRESS=0x70997970c51812dc3a010c7d01b50e0d17dc79c8
+      - DEPLOYER_PRIVATE_KEY=<deployer-private-key>
```
Comment (bap2pecs) [inline docker-compose.yml:412]: does this have to be the first key derived from the MNEMONIC above in the other container?

**Review by bap2pecs — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 6 above)
Comment (lesterli) [reply docker-compose.yml:412]: yes, from the `MNEMONIC` in `cdk-validium-deploy-sepolia`

**Review by bap2pecs — APPROVED** [review]
lgtm. one more thing, could you create a new branch `develop` and merge this PR to it instead of `main?

**Comment by bap2pecs** [comment]
closed in favor of https://github.com/Snapchain/zkValidium-quickstart/pull/6

**Pull request closed without merging**